design-documents/graph-ql/coverage/catalog/configurable-options-selection.graphqls

```diff
@@ -1,21 +1,29 @@
 type ConfigurableProduct {
-    configurable_options_selection_metadata(selectedConfigurableOptionValues: [ID!]): ConfigurableOptionsSelectionMetadata @doc(description: "Metadata for the specified configurable options selection")
+    configurable_product_options_selection(configurableOptionValues: [ID!]): ConfigurableProductOptionsSelection @doc(description: "Specified configurable product options selection")
+    variants: [ConfigurableVariant] @deprecated(reason: "Use configurable_options_selection_metadata instead.") @doc(description: "An array of variants of products") @resolver(class: "Magento\\ConfigurableProductGraphQl\\Model\\Resolver\\ConfigurableVariant")
+    configurable_options: [ConfigurableProductOptions] @deprecated(reason: "Use configurable_options_selection_metadata instead.") @doc(description: "An array of linked simple product items") @resolver(class: "Magento\\ConfigurableProductGraphQl\\Model\\Resolver\\Options")
 }
 
-type ConfigurableOptionsSelectionMetadata @doc(description: "Metadata corresponding to the configurable options selection.")
+type ConfigurableProductOptionsSelection @doc(description: "Metadata corresponding to the configurable options selection.")
 {
-    options_available_for_selection: [ConfigurableOptionAvailableForSelection!] @doc(description: "Configurable options available for further selection based on current selection.")
+    configurable_options: [ConfigurableProductOption!] @doc(description: "Configurable options available for further selection based on current selection.")
     media_gallery: [MediaGalleryInterface!] @doc(description: "Product images and videos corresponding to the specified configurable options selection.")
     variant: SimpleProduct @doc(description: "Variant represented by the specified configurable options selection. It is expected to be null, until selections are made for each configurable option.")
 }
 
-type ConfigurableOptionAvailableForSelection @doc(description: "Configurable option available for further selection based on current selection.") {
-    option_value_uids: [ID!]! @doc(description: "Configurable option values available for further selection.")
-    attribute_code: String! @doc(description: "Attribute code that uniquely identifies configurable option.")
+type ConfigurableProductOption {
+    uid: ID!
```

Comment:
what data will be present in uid?

In ConfigurableVariant we had a link to Product which is a representation of Simple product for specific variation. 
How can I receive Product ID with new approach to add the product to the Cart?

Reply:
We don't need id of the product for adding to cart, but ConfigurableProductOptionsSelection.variant is the selected product.

Reply:
@mslabko For existing mutation addConfigurableProductsToCart, the variant can be added with the variant_sku. For the new single mutation for all products, addProductsToCart, the uid on the ConfigurableProductOptionValue can be used for adding the variant without variant_sku. 
Like @melnikovi mentioned this uid is added for improving client side caching.